[StarOS] 'Cannot open null host' error

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
````sql
mysql> show backends;
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
| BackendId | IP            | HeartbeatPort | BePort | HttpPort | BrpcPort | LastStartTime       | LastHeartbeat       | Alive | SystemDecommissioned | ClusterDecommissioned | TabletNum | DataUsedCapacity | AvailCapacity | TotalCapacity | UsedPct | MaxDiskUsedPct | ErrMsg | Version         | Status                                                 | DataTotalCapacity | DataUsedPct | CpuCores | NumRunningQueries | MemUsedPct | CpuUsedPct | StarletPort | WorkerId |
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
| 10005     | 172.26.92.200 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:43:36 | 2023-02-16 16:09:44 | true  | false                | false                 | 36        | 2.980 MB         | 893.324 GB    | 1.968 TB      | 55.68 % | 55.68 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:09:37"} | 893.327 GB        | 0.00 %      | 16       | 0                 | 2.22 %     | 0.0 %      | 8167        | 1        |
| 10114     | 172.26.92.201 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:42:51 | 2023-02-16 16:09:44 | true  | false                | false                 | 41        | 3.980 MB         | 1.283 TB      | 1.968 TB      | 34.81 % | 34.81 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:08:51"} | 1.283 TB          | 0.00 %      | 16       | 0                 | 2.44 %     | 0.0 %      | 8167        | 2        |
| 10115     | 172.26.92.202 | 9354          | 9360   | 8338     | 8360     | 2023-02-16 15:42:56 | 2023-02-16 16:09:44 | true  | false                | false                 | 36        | 2.978 MB         | 1.229 TB      | 1.968 TB      | 37.54 % | 37.54 %        |        | UNKNOWN-46580be | {"lastSuccessReportTabletsTime":"2023-02-16 16:08:57"} | 1.229 TB          | 0.00 %      | 16       | 0                 | 2.30 %     | 0.0 %      | 8167        | 3        |
+-----------+---------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+-----------+------------------+---------------+---------------+---------+----------------+--------+-----------------+--------------------------------------------------------+-------------------+-------------+----------+-------------------+------------+------------+-------------+----------+
3 rows in set (0.01 sec)

mysql> use tn_test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+-------------------+
| Tables_in_tn_test |
+-------------------+
| detail_demo       |
+-------------------+
1 row in set (0.00 sec)

mysql> CREATE TABLE `primary_table_without_null` (
        ->    `k1` date NOT NULL,
        ->    `k2` datetime NOT NULL,
        ->    `k3` string NOT NULL,
        ->    `k4` string NOT NULL,
        ->    `k5` boolean NOT NULL,
        ->    `v1` tinyint NOT NULL,
        ->    `v2` smallint NOT NULL,
        ->    `v3` int NOT NULL,
        ->    `v4` bigint NOT NULL,
        ->    `v5` largeint NOT NULL,
        ->    `v6` float NOT NULL,
        ->    `v7` double NOT NULL,
        ->    `v8` decimal(27,9) NOT NULL
        ->  )
        ->  PRIMARY KEY(`k1`, `k2`)
        ->  COMMENT "OLAP"
        ->  DISTRIBUTED BY HASH(`k1`, `k2`) BUCKETS 3
        ->  PROPERTIES (
        ->    "replication_num" = "3",
        ->    "enable_persistent_index" = "true",
        ->    "storage_format" = "v2"
        ->  );
ERROR 1064 (HY000): Cannot open null host.
mysql> show backends;
ERROR 1064 (HY000): Cannot open null host.
mysql> show backends;
````

### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
46580be
```

fe.warn.log
```
2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
Caused by: java.io.EOFException
        at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
        at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
        ... 5 more
```

> fe.warn.log
> 
> ```
> 2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
> com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
>         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
>         at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
>         at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
> Caused by: java.io.EOFException
>         at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
>         at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
>         at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
>         at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
>         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
>         ... 5 more
> ```

This is the root cause, a broken journal caused the FE node unable to receive the leader switch journal, so the FE node dit not know which FE is leader, i.e, the saved IP address of the leader node is a null.

> > fe.warn.log
> > ```
> > 2023-02-16 16:19:49,943 ERROR (replayer|81) [GlobalStateMgr$5.runOneCycle():1828] replayer thread catch an exception when replay journal 107935.
> > com.starrocks.journal.JournalException: fail to read journal entity key=107935, data=<DatabaseEntry offset="0" size="5613" data="39 18 0 0 0 3 0 0 0 0 0 1 17 214 0 0 0 0 0 1 17 238 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 18 15 0 0 0 13 112 50 48 50 48 48 54 95 55 48 49 53 56 0 0 0 6 78 79 82 77 65 76 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 0 0 0 0 0 1 17 239 0 0 0  ... 5513 bytes not shown "/>
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:257) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:295) ~[starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1954) ~[starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1819) [starrocks-fe.jar:?]
> >         at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
> >         at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1884) [starrocks-fe.jar:?]
> > Caused by: java.io.EOFException
> >         at java.io.DataInputStream.readFully(DataInputStream.java:197) ~[?:1.8.0_252]
> >         at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Replica.readFields(Replica.java:571) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Replica.read(Replica.java:582) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.LocalTablet.readFields(LocalTablet.java:388) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.LocalTablet.read(LocalTablet.java:404) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.MaterializedIndex.readFields(MaterializedIndex.java:297) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.MaterializedIndex.read(MaterializedIndex.java:308) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Partition.readFields(Partition.java:419) ~[starrocks-fe.jar:?]
> >         at com.starrocks.catalog.Partition.read(Partition.java:372) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.PartitionPersistInfo.readFields(PartitionPersistInfo.java:133) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.PartitionPersistInfo.read(PartitionPersistInfo.java:126) ~[starrocks-fe.jar:?]
> >         at com.starrocks.persist.AddPartitionsInfo.read(AddPartitionsInfo.java:49) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:278) ~[starrocks-fe.jar:?]
> >         at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:251) ~[starrocks-fe.jar:?]
> >         ... 5 more
> > ```
> 
> This is the root cause, a broken journal caused the FE node unable to receive the leader switch journal, so the FE node dit not know which FE is leader, i.e, the saved IP address of the leader node is a null.

Here is one way to reproduce the problem:
```
CREATE TABLE t0(c0 INT) ENGINE=STARROCKS DUPLICATE KEY(c0) DISTRIBUTED BY HASH(c0);
INSERT INTO t0 VALUES(1);
INSERT OVERWRITE t0 VALUES(2);
```

--- a/fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java
+++ b/fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java
@@ -313,6 +313,8 @@ public class InsertOverwriteJobRunner {
                     invertedIndex.markTabletForceDelete(tabletId);
                 }
 
+                GlobalStateMgr.getCurrentColocateIndex().updateLakeTableColocationInfo(targetTable);
+
                 InsertOverwriteStateChangeInfo info = new InsertOverwriteStateChangeInfo(job.getJobId(), job.getJobState(),
                         InsertOverwriteJobState.OVERWRITE_SUCCESS, job.getSourcePartitionIds(), job.getTmpPartitionIds());
                 GlobalStateMgr.getCurrentState().getEditLog().logInsertOverwriteStateChange(info);

--- a/fe/fe-core/src/main/java/com/starrocks/load/PartitionUtils.java
+++ b/fe/fe-core/src/main/java/com/starrocks/load/PartitionUtils.java
@@ -27,7 +27,11 @@ import com.starrocks.catalog.RangePartitionInfo;
 import com.starrocks.catalog.Table;
 import com.starrocks.common.DdlException;
 import com.starrocks.persist.AddPartitionsInfo;
+import com.starrocks.persist.AddPartitionsInfoV2;
 import com.starrocks.persist.PartitionPersistInfo;
+import com.starrocks.persist.PartitionPersistInfoV2;
+import com.starrocks.persist.RangePartitionPersistInfo;
+import com.starrocks.persist.SinglePartitionPersistInfo;
 import com.starrocks.server.GlobalStateMgr;
 
 import java.util.List;
@@ -55,13 +59,15 @@ public class PartitionUtils {
                     .map(id -> targetTable.getPartition(id)).collect(Collectors.toList());
             PartitionInfo partitionInfo = targetTable.getPartitionInfo();
             List<PartitionPersistInfo> partitionInfoList = Lists.newArrayListWithCapacity(newTempPartitions.size());
+            List<PartitionPersistInfoV2> partitionInfoV2List = Lists.newArrayListWithCapacity(newTempPartitions.size());
             for (int i = 0; i < newTempPartitions.size(); i++) {
                 targetTable.addTempPartition(newTempPartitions.get(i));
                 long sourcePartitionId = sourcePartitions.get(i).getId();
                 partitionInfo.addPartition(newTempPartitions.get(i).getId(),
                         partitionInfo.getDataProperty(sourcePartitionId),
                         partitionInfo.getReplicationNum(sourcePartitionId),
-                        partitionInfo.getIsInMemory(sourcePartitionId));
+                        partitionInfo.getIsInMemory(sourcePartitionId),
+                        partitionInfo.getStorageCacheInfo(sourcePartitionId));
                 Partition partition = newTempPartitions.get(i);
                 // range is null for UNPARTITIONED type
                 Range<PartitionKey> range = null;
@@ -71,17 +77,41 @@ public class PartitionUtils {
                             rangePartitionInfo.getRange(sourcePartitionId));
                     range = rangePartitionInfo.getRange(partition.getId());
                 }
-                PartitionPersistInfo info =
-                        new PartitionPersistInfo(db.getId(), targetTable.getId(), partition,
-                                range,
-                                partitionInfo.getDataProperty(partition.getId()),
+                if (targetTable.isLakeTable()) {
+                    PartitionPersistInfoV2 info = null;
+                    if (range != null) {
+                        info = new RangePartitionPersistInfo(db.getId(), targetTable.getId(),
+                                partition, partitionInfo.getDataProperty(partition.getId()),
                                 partitionInfo.getReplicationNum(partition.getId()),
-                                partitionInfo.getIsInMemory(partition.getId()),
-                                true);
-                partitionInfoList.add(info);
+                                partitionInfo.getIsInMemory(partition.getId()), true,
+                                range, partitionInfo.getStorageCacheInfo(partition.getId()));
+                    } else {
+                        info = new SinglePartitionPersistInfo(db.getId(), targetTable.getId(),
+                                partition, partitionInfo.getDataProperty(partition.getId()),
+                                partitionInfo.getReplicationNum(partition.getId()),
+                                partitionInfo.getIsInMemory(partition.getId()), true,
+                                partitionInfo.getStorageCacheInfo(partition.getId()));
+                    }
+                    partitionInfoV2List.add(info);
+                } else {
+                    PartitionPersistInfo info =
+                            new PartitionPersistInfo(db.getId(), targetTable.getId(), partition,
+                                    range,
+                                    partitionInfo.getDataProperty(partition.getId()),
+                                    partitionInfo.getReplicationNum(partition.getId()),
+                                    partitionInfo.getIsInMemory(partition.getId()),
+                                    true);
+                    partitionInfoList.add(info);
+                }
             }
-            AddPartitionsInfo infos = new AddPartitionsInfo(partitionInfoList);
-            GlobalStateMgr.getCurrentState().getEditLog().logAddPartitions(infos);
+
+            if (targetTable.isLakeTable()) {
+                AddPartitionsInfoV2 infos = new AddPartitionsInfoV2(partitionInfoV2List);
+                GlobalStateMgr.getCurrentState().getEditLog().logAddPartitions(infos);
+            } else {
+                AddPartitionsInfo infos = new AddPartitionsInfo(partitionInfoList);
+                GlobalStateMgr.getCurrentState().getEditLog().logAddPartitions(infos);
+            }
         } finally {
             db.writeUnlock();
         }

--- a/fe/fe-core/src/main/java/com/starrocks/persist/PartitionPersistInfoV2.java
+++ b/fe/fe-core/src/main/java/com/starrocks/persist/PartitionPersistInfoV2.java
@@ -27,7 +27,7 @@ import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class PartitionPersistInfoV2 implements Writable {
+public abstract class PartitionPersistInfoV2 implements Writable {
 
     @SerializedName("dbId")
     private Long dbId;

--- a/fe/fe-core/src/main/java/com/starrocks/persist/SinglePartitionPersistInfo.java
+++ b/fe/fe-core/src/main/java/com/starrocks/persist/SinglePartitionPersistInfo.java
@@ -1,0 +1,29 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.persist;
+
+import com.starrocks.catalog.DataProperty;
+import com.starrocks.catalog.Partition;
+import com.starrocks.lake.StorageCacheInfo;
+
+// UNPARTITIONED type
+public class SinglePartitionPersistInfo extends PartitionPersistInfoV2 {
+
+    public SinglePartitionPersistInfo(Long dbId, Long tableId, Partition partition,
+                                      DataProperty dataProperty, short replicationNum, boolean isInMemory,
+                                      boolean isTempPartition, StorageCacheInfo storageCacheInfo) {
+        super(dbId, tableId, partition, dataProperty, replicationNum, isInMemory, isTempPartition, storageCacheInfo);
+    }
+}

--- a/fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java
+++ b/fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java
@@ -101,6 +101,7 @@ import com.starrocks.load.loadv2.SparkLoadJob.SparkLoadJobStateUpdateInfo;
 import com.starrocks.persist.ListPartitionPersistInfo;
 import com.starrocks.persist.PartitionPersistInfoV2;
 import com.starrocks.persist.RangePartitionPersistInfo;
+import com.starrocks.persist.SinglePartitionPersistInfo;
 import com.starrocks.privilege.CatalogPEntryObject;
 import com.starrocks.privilege.DbPEntryObject;
 import com.starrocks.privilege.FunctionPEntryObject;
@@ -218,7 +219,8 @@ public class GsonUtils {
     private static final RuntimeTypeAdapterFactory<PartitionPersistInfoV2> PARTITION_PERSIST_INFO_V_2_ADAPTER_FACTORY
             = RuntimeTypeAdapterFactory.of(PartitionPersistInfoV2.class, "clazz")
             .registerSubtype(ListPartitionPersistInfo.class, ListPartitionPersistInfo.class.getSimpleName())
-            .registerSubtype(RangePartitionPersistInfo.class, RangePartitionPersistInfo.class.getSimpleName());
+            .registerSubtype(RangePartitionPersistInfo.class, RangePartitionPersistInfo.class.getSimpleName())
+            .registerSubtype(SinglePartitionPersistInfo.class, SinglePartitionPersistInfo.class.getSimpleName());
 
     private static final RuntimeTypeAdapterFactory<CatalogRecycleBin.RecyclePartitionInfoV2>
             RECYCLE_PARTITION_INFO_V_2_ADAPTER_FACTORY

--- a/fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java
+++ b/fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java
@@ -1444,8 +1444,12 @@ public class LocalMetastore implements ConnectorMetadata {
             } else if (partitionInfo.isRangePartition()) {
                 ((RangePartitionInfo) partitionInfo).unprotectHandleNewSinglePartitionDesc(
                         info.asRangePartitionPersistInfo());
+            } else if (partitionType == PartitionType.UNPARTITIONED) {
+                // insert overwrite job will create temp partition and replace the single partition.
+                partitionInfo.addPartition(partition.getId(), info.getDataProperty(), info.getReplicationNum(),
+                        info.isInMemory(), info.getStorageCacheInfo());
             } else {
-                throw new DdlException("Only support adding partition to range/list partitioned table");
+                throw new DdlException("Unsupported partition type: " + partitionType.name());
             }
 
             if (!isCheckpointThread()) {

